.github/workflows/ci-test-ginkgo.yml

```diff
@@ -77,18 +77,21 @@ jobs:
           path: |
             /tmp/kubearmor/
             /tmp/kubearmor.*
-
-      - name: Measure code coverage
-        if: ${{ always() }}
-        run: |
-          go install github.com/modocache/gover@latest
-          gover
-          go tool cover -func=gover.coverprofile
-        working-directory: KubeArmor
-        env:
-          GOPATH: /home/runner/go
-
-      - uses: codecov/codecov-action@v3
-        if: ${{ always() }}
-        with:
-          files: ./KubeArmor/gover.coverprofile
+  codecoverage:
+    needs: [build]
+    - name: Measure code coverage / ${{ matrix.os }} / ${{ matrix.runtime }} 
+    runs-on: ${{ matrix.os }}
+    env:
+      RUNTIME: ${{ matrix.runtime }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04]
+        runtime: ["docker", "containerd", "crio"]
+    steps: 
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
```

Comment:
We are trying to download the artifacts here. But do they exist? I belive we need to upload them in the jobs. At the same time this shouldn't run on each workflow. Ideally this should be seperate workflow which runs after all of the tests have run.